src/components/Main/Results/DeterministicLinePlot.tsx

```diff
@@ -92,21 +92,40 @@ export function DeterministicLinePlot({
 
   const nonEmptyCaseCounts = caseCounts?.filter((d) => d.cases || d.deaths || d.icu || d.hospitalized)
 
-  const caseStep = 3
+  const caseBaseStep = Math.floor(params.epidemiological.infectiousPeriod)
+  const caseDelta = params.epidemiological.infectiousPeriod - caseBaseStep
+
   // this currently relies on there being data for every day. This should be
   // the case given how the data are parsed, but would be good to put in a check
-  const newCases = (cc: EmpiricalData, i: number) => {
-    if (i >= caseStep && cc[i].cases !== null && cc[i - caseStep].cases !== null) {
-      return verifyPositive((cc[i].cases as number) - (cc[i - caseStep].cases as number))
-    }
-    return undefined
-  }
+  const newCases =
+    caseDelta === 0
+      ? (cc: EmpiricalData, i: number) => {
+          if (i >= caseBaseStep && cc[i].cases !== null && cc[i - caseBaseStep].cases !== null) {
+            return verifyPositive((cc[i].cases as number) - (cc[i - caseBaseStep].cases as number))
+          }
+          return undefined
+        }
+      : (cc: EmpiricalData, i: number) => {
+          if (
+            i >= caseBaseStep + 1 &&
+            cc[i].cases !== null &&
+            cc[i - caseBaseStep + 1].cases !== null &&
+            cc[i - caseBaseStep].cases !== null
+          ) {
+            return verifyPositive(
+              (cc[i].cases as number) -
+                ((1 - caseDelta) * (cc[i - caseBaseStep].cases as number) +
+                  caseDelta * (cc[i - caseBaseStep + 1].cases as number)),
+            )
+          }
+          return undefined
+        }
```

Comment:
I don't know how to read this. Can we please agree on limiting ternary operators to one-liners only?

As there are 4 booleans combined in an if statement, it makes me think that this particular check does something very specific. But I cannot immediately see what. Can we refactor it into a named function, so that the name would give me a hint?

Please refactor all duplicated statements like `i - caseBaseStep` and `cc[i - caseBaseStep]` into well-named variables.

Why type-casting (`as number`) is needed? I am trying to introduce type checks into the codebase and casts go against that. After cast is written it is extremely hard to find them. I'd prefer you leave type errors that I can see and fix later, or even better that you fix them right away.

Do we really need to produce 2 arrow functions conditionally? Can this condition be moved inside the function?
Can then this function be a normal named `function` in the outside scope, so that the component contains less fluff?

Reply:
I agree this is too convoluted. But it should be pretty straightforward to break into more explicit code. 

One other thing: shouldn't `i - caseBaseStep + 1` be `i - caseBaseStep - 1`?

Reply:
In my opinion producing one or the other function in one branch statement is cleaner than checking this everytime the function is called within the body. But I've made the required changes. The code should be much more obvious now